### DISCUSSION TITLE
How don't share LocalStorage on the same subdomain

### DISCUSSION THREAD

**Discussion by devascolades** [vuejs/pinia#2470] [Help and Questions] [answered]
Hello,

We have completely different websites on the same subdomain :

sub.domain.com/site1
sub.domain.com/site2

The problem is that the localstorage in the browser (where pinia saves the data) automatically shares the data as if it were a single website (sub.domain.com) , even though they are 2 different sites.

Example : 
If I connect to sub.domain.com/site1 (pinia save the username) and after go to sub.domain.com/site2 without authentification,   sub.domain.com/site2 will retrieve information from sub.domain.com/site1 (username)

export const piniaStore = defineStore('default-store', {
  state: () => ({
    username : null,
  }),

Is it possible to differentiate the localstorage between these 2 websites ?

**Comment by posva** [comment, marked as answer]
You cannot use Local Storage for this, it doesn't have that functionality. You could use cookies.
Another way would be to host the page in different subdomains

**Comment by SultanDino** [comment]
Had a similar discussion/issue.
You could set a prefix to the store name. The prefix should be site1 or site2. That way you will have 2 separate website states in same localStorage
